pkg/cluster/controllers/storage/healthchecker.go

```diff
@@ -25,10 +33,71 @@ type HealthChecker struct{}
 
 var _ lcm.HealthChecker = &HealthChecker{}
 
-// TODO: Implement me
 // CheckHealth implements lcm.HealthChecker interface for checking health of the storage.
-func (hc *HealthChecker) CheckHealth(ctx context.Context, svc *lcm.ServiceConfig, options ...lcm.Option) (*lcm.CheckResponse, error) {
-	return &lcm.CheckResponse{
+func (c *HealthChecker) CheckHealth(ctx context.Context, svc *lcm.ServiceConfig, options ...lcm.Option) (*lcm.CheckResponse, error) {
+	checkOpts := &lcm.CheckOptions{}
+
+	for _, o := range options {
+		o(checkOpts)
+	}
+
+	switch checkOpts.StorageDriver {
+	case goharborv1.S3DriverName:
+		return S3StorageHealthCheck(ctx, svc, checkOpts)
+	case goharborv1.SwiftDriverName:
+		return SwiftStorageHealthCheck(ctx, svc, checkOpts)
+	case goharborv1.FileSystemDriverName:
+		return &lcm.CheckResponse{
```

Comment:
maybe we can check whether folder exists

Reply:
I don't think it's necessary, because the PersistentVolume mounted will always create the folder.